python/src/scipp/compat/mantid.py

```diff
@@ -161,9 +161,127 @@ def convert_EventWorkspace_to_dataset(ws, load_pulse_times, EventType):
     return sc.DataArray(**coords_labs_data)
 
 
+def convert_TableWorkspace_to_dataset(ws, error_connection=None):
+    """
+    Converts from a Mantid TableWorkspace to a scipp dataset. A TableWorkspace
+    have optional declaration of connection between data and error using the
+    plottype which can be none, X, Y, Z, Xerr, Yerr. This function collects
+    the data and its error into a single scipp variable if there is no
+    ambiguity in what column contains the error for what data column. This
+    means a maximum of one Xerr and one Yerr column. For more advanced
+    TableWorkspaces, the user can specify the connection between data and error
+    manually using the error_connections keyword argument, which overwrites
+    automatic assignment.
+
+    Parameters
+    ----------
+        ws : Mantid TableWorkspace
+            Mantid TableWorkspace to be converted into scipp dataset
+
+    Keyword arguments
+    -----------------
+        error_connection : Dict
+            Dict with data column names as keys to names of their error column
+    """
+
+    # Extract information from workspace
+    n_columns = ws.columnCount()
+    columnNames = ws.getColumnNames()  # list of names matching each column
+    columnTypes = ws.columnTypes()  # list of types matching each column
+
+    if error_connection is None:
+        # Generate error connection
+        error_connection = {}
+        # Dictionary with data column keys for names of their error columns
+        # Uses the plottype that is built into TableWorkspace
+        plottype_list = []
+        for i in range(n_columns):
+            plottype_list.append(ws.getPlotType(i))
+
+        # Relevant plottype indices
+        X_index = 1
+        Xerr_index = 4
+        Y_index = 2
+        Yerr_index = 5
```

Comment:
I am thinking we can use this information to a fuller extent. From the Mantid docs:

```
/// Set plot type where
/// None = 0 (means it has specifically been set to 'no plot type')
/// NotSet = -1000 (this is the default and means plot style has not been set)
/// X = 1, Y = 2, Z = 3, xErr = 4, yErr = 5, Label = 6
/// @param t plot type as defined above
```

I propose:
- X (and xErr) -> create `coord` "column". Not sure what to do about the name. Can we have special names that map to dimension names?
- Label -> create `labels` "column".
- everything else -> create normal data columns.

When set up like this the existing `scipp` plotting functionality should be able to generate reasonable plots for a number of cases. As it is now (no coords) it would not, I think.

Reply:
It was decided to not use the plottype information from the TableWorkspace, so this code was deleted. The plottype information is simply not used in mantid, so there is no reason to support it from scipp, even though the scheme suggested here would have been neat.